pkg/nuget/lock/parse.go

```diff
@@ -60,9 +62,34 @@ func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 				},
 			}
 			libs = append(libs, lib)
+
+			depId := utils.PackageID(packageName, packageContent.Resolved)
```

Comment:
You also need to pass the ID to `types.Library` here.
https://github.com/aquasecurity/go-dep-parser/blob/5545d9feecb875ce2c9f0ea56d7a9a726ed06438/pkg/nuget/lock/parse.go#L54-L63